keyring.php

```diff
@@ -130,6 +130,10 @@ function() {
 	static function request_handlers() {
 		global $current_user;
 
+		if ( ! empty( $_REQUEST['state'] ) ) {
```

Comment:
I'm concerned that this method is called on every admin request. Should we create another PR that tidies that up and check we're on the keyring tools page? Or if we end up on different pages, adding a `keyring` parameter to avoid any unnecessary processing.

Reply:
The only problem is that in the case of the instagram callback we don't have any query params to order to indicate that keyring tools page. This method already runs on every admin call anyway as it currently doesn't check for the tools keyring param, so I don't think we are introducing any extra overhead, so I think you are correct that we should look at this in a separate PR if you think it is worth following up.